sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -300,10 +300,21 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder {
    * Create a [[DescribeTableCommand]] logical plan.
    */
   override def visitDescribeTable(ctx: DescribeTableContext): LogicalPlan = withOrigin(ctx) {
-    // Describe column are not supported yet. Return null and let the parser decide
-    // what to do with this (create an exception or pass it on to a different system).
     if (ctx.describeColName != null) {
-      null
+      if (ctx.partitionSpec != null) {
+        throw new ParseException("DESC TABLE COLUMN for a specific partition is not supported", ctx)
+      } else {
+        val columnName = ctx.describeColName.getText
```

Comment:
the parser rule for the column name here:
```
describeColName
    : identifier ('.' (identifier | STRING))*
    ;
```
can we just make it `identifier`? "a.b" should refer to a column named "a.b", or the inner field "b" from column "a"? let's check with other databases.